image/templates/helm/stackrox-central/config/centraldb/postgresql.conf.default

```diff
@@ -26,4 +26,32 @@ lc_numeric = 'en_US.utf8'		# locale for number formatting
 lc_time = 'en_US.utf8'			# locale for time formatting
 
 default_text_search_config = 'pg_catalog.english'
-shared_preload_libraries = 'pg_stat_statements'	# StackRox customized
\ No newline at end of file
+shared_preload_libraries = 'pg_stat_statements'	# StackRox customized
+
+# Logging. For more details, see
+# https://www.postgresql.org/docs/current/runtime-config-logging.html
+
+# It's convenienv for troubleshooting to log which client has connected and
+# disconnected and when.
+log_connections = 'on'
+log_disconnections = 'on'
+
+# Checkpoints might affect IO throughput.
+log_checkpoints = 'on'
+
+# Make excessive locking visible, since it could affect query latency.
+log_lock_waits = 'on'
+
+# It's useful for troubleshooting to log any query, that took longer than
+# 500ms. Just in case if there is any sensitive information in the query
+# parameters, do not log them, only the query itself.
+log_min_duration_statement = 500
+log_parameter_max_length = 0
```

Comment:
Any concerns about this making the problem worse when those customer environments start to get slow?

Any concerns about leaking our queries to the customer? 

As an aside do have a way to get this today by having the customer set the flags in https://github.com/stackrox/stackrox/blob/master/pkg/env/postgres_query_tracer.go.  At which case those queries will be logged to central.

Reply:
> Any concerns about this making the problem worse when those customer environments start to get slow?

Not really, the overhead here is only to put slow queries in the log which might be a problem only for queries with large number of parameters. In my testing environment ingesting chunks of 10k size I haven't noticed any significant impact.

> Any concerns about leaking our queries to the customer?

What would be  a concern here? AFAICT the queries are not secret :)

> As an aside do have a way to get this today by having the customer set the flags

This will measure the latency of the DB + network, which would be different from what would be logged here (only the DB time).